src/modules/job-manager/event.c

```diff
@@ -672,12 +673,12 @@ int event_job_post_pack (struct event *event,
         goto error;
     if (!(entry = eventlog_entry_vpack (timestamp, name, context_fmt, ap)))
         return -1;
+    if (event_batch_process_event_entry (event, job, name, entry) < 0)
+        goto error;
     if (event_job_update (job, entry) < 0) // modifies job->state
         goto error;
     if (event_batch_commit_event (event, job, entry) < 0)
         goto error;
-    if (event_batch_process_event_entry (event, job, name, entry) < 0)
-        goto error;
```

Comment:
This reordering probably doesn't belong in this commit?  It should probably have a commit message of its own since the change has a non-obvious (to me) rationale.